shared/src/main/scala/coop/rchain/dag/DagReaderSyntax.scala

```diff
@@ -0,0 +1,31 @@
+package coop.rchain.dag
+
+import cats.effect.Concurrent
+import coop.rchain.dag.DagOps.bfTraverseF
+import coop.rchain.shared.syntax._
+import cats.syntax.all._
+import fs2.Stream
+
+trait DagReaderSyntax {
+  implicit final def syntaxDagReader[F[_], A](
+      dag: DagReader[F, A]
+  ): DagReaderOps[F, A] = new DagReaderOps[F, A](dag)
+}
+
+final class DagReaderOps[F[_], A](
+    private val dag: DagReader[F, A]
+) {
+
+  def gatherAncestors(start: A, stopCondition: A => F[Boolean])(
```

Comment:
Type parameter `A` can be defined on the method itself. In that way it can be resolved when method is called and not when implicit (syntax) is resolved.
In general type parameters (the same as value level variables) should be defined as late as possible.

P.S.
Class with extension methods should extend `AnyVal` to escape boxing on each call.
https://typelevel.org/cats/guidelines.html
> Ops classes should be marked final and extend AnyVal, to take full advantage of inlining and prevent unnecessary allocations.